### PR TITLE
Implement generic unknown thing UI.

### LINKED ISSUE #511: Unknown Thing UI
As a user I would like to be able to add a thing to the gateway, view and set its properties, and request actions, even if it doesn't conform to one of the standard thing types.

Currently the UI only lets you interact with things which conform to a standard Thing type. As people create custom things users should be able to add those things to the gateway and interact with them via a generic UI.

Mockups in progress...

### PR DESCRIPTION
**Pull request by mrstegeman**
Fixes #511

### CODE REVIEW

File: static/js/unknown-thing.js [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+  }
+};
+
+/**
+ * Update the display for the provided property.
+ * @param {string} name - name of the property
+ * @param {*} value - value of the property
+ */
+UnknownThing.prototype.updateProperty = function(name, value) {
+  console.log(name);
```
Comment (hobinjk) [inline static/js/unknown-thing.js]: nit: unnecessary console.log

File: static/js/unknown-thing.js [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+/* globals Thing, ThingDetailLayout */
+
+/**
+ * UnknownThing Constructor (extends Thing).
+ *
+ * @param Object description Thing description object.
+ */
+var UnknownThing = function(description, format) {
+  this.displayedProperties = {};
+  if (description.properties) {
```
Comment (hobinjk) [inline static/js/unknown-thing.js]: This should also include if (format === 'htmlDetail') or the updateProperty method later on might call update() on an unattached detail

Comment (mrstegeman) [reply static/js/unknown-thing.js]: All of the `update()` methods already handle that case.

Comment (mrstegeman) [reply static/js/unknown-thing.js]: Doing that also ends up breaking rendering of the HTML views, as the code has to look up if there are additional properties or not to justify a full expanded view.

Comment (hobinjk) [reply static/js/unknown-thing.js]: Ah, sounds good. Thanks for the explanation!

**Pull request merged**